service/host.go

```diff
@@ -85,26 +95,43 @@ func (uis *UIServer) hostPage(w http.ResponseWriter, r *http.Request) {
 	uis.render.WriteResponse(w, http.StatusOK, struct {
 		Events      []event.EventLogEntry
 		Host        *host.Host
+		Permissions gimlet.Permissions
 		RunningTask *task.Task
 		Containers  []host.Host
 		ViewData
-	}{events, h, runningTask, containers, uis.GetCommonViewData(w, r, false, true)},
+	}{events, h, permissions, runningTask, containers, uis.GetCommonViewData(w, r, false, true)},
 		"base", "host.html", "base_angular.html", "menu.html")
 }
 
 func (uis *UIServer) hostsPage(w http.ResponseWriter, r *http.Request) {
+	u := MustHaveUser(r)
+
 	includeSpawnedHosts, _ := strconv.ParseBool(r.FormValue(IncludeSpawnedHosts))
 	hosts, err := getHostsData(includeSpawnedHosts)
 	if err != nil {
 		uis.LoggedError(w, r, http.StatusInternalServerError, err)
 		return
 	}
 
+	permittedHosts := &hostsData{}
+	for i := range hosts.Hosts {
+		opts := gimlet.PermissionOpts{Resource: hosts.Hosts[i].Host.Distro.Id, ResourceType: evergreen.DistroResourceType}
+		permissions, err := rolemanager.HighestPermissionsForRoles(u.Roles(), evergreen.GetEnvironment().RoleManager(), opts)
```

Comment:
Similar to the distros page, we should avoid doing a query in a loop like this. Maybe we should have a public function that does this check for multiple resources